main.tf

```diff
@@ -37,11 +37,19 @@ resource "azurerm_key_vault" "keyvault" {
 
 # role assignments
 resource "azurerm_role_assignment" "current" {
+  count                = try(var.vault.current_identity_as_admin, true) ? 1 : 0
```

Comment:
i should not use count, there are better alternatives and it might give complications

Reply:
can you provide a better alternative? we have used count here to make the role assignment optional.